### PR TITLE
Running the tests of some packages that use Solid

### LINKED ISSUE #358: [Tests] [Suggestion] Running the tests of some packages that use Solid
**Describe the bug**
Coming from all the issues I have had since updating Solid to 0.24 (reported in #335), I decided to give you this suggestion:

Should Solid run the tests for some of the packages that use Solid? This will avoid this kind of breakage from happening again, and this can be used actually as unit tests for the framework itself under real conditions.

As an example, Solid-Simple-Table has 400 lines of tests testing everything like rendering, clicking, etc. This can be useful for Solid itself to test if it is functioning properly.
https://github.com/aminya/solid-simple-table/blob/master/test/SimpleTable.test.tsx
https://github.com/aminya/solid-simple-table/blob/master/demo/simple/index.tsx
https://github.com/aminya/solid-simple-table/blob/master/demo/complex/index.tsx
https://github.com/aminya/solid-simple-table/blob/master/demo/variable-rows/index.tsx

**Solution Prototype**

I already do this for one of my packages that is used by many other packages. Here is a sample script for downloading the tests of other packages and running them:
https://github.com/atom-community/eslint-config-atomic/blob/master/spec/test.js

I can make a PR adding that if you are interested. 

**Additional context**
See #335

### PR DESCRIPTION
**Pull request by aminya**
Fixes #358

Gitly doesn't download on my local machine. Not sure if I am being rate-limited.

### CODE REVIEW

File: packages/solid/test/downloaded.spec.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  rm("-rf", packedPkg);
+  exec("npm pack", { cwd: packageRoot, fatal: true });
+  return packedPkg;
+}
+
+function packageExists(packageRoot: string) {
+  return existsSync(join(packageRoot, "package.json"));
+}
+
+describe("Downloaded tests", () => {
```
Comment (aminya) [inline packages/solid/test/downloaded.spec.ts:65]: I can make all the above code an npm package to abstract things. This looks like a useful tool to have.

File: packages/solid/test/downloaded.spec.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  // The repositories to run the tests for
+  const testRepos: TestRepo[] = [makeTestRepo("aminya/solid-simple-table")];
```
Comment (aminya) [inline packages/solid/test/downloaded.spec.ts:67]: Once we get the code working, we can add more repositories here.

**Comment by ryansolid** [comment]
This is awesome. I think I might move some stuff around to make this a private like test package in the repo. But this looks great. Might take me a little bit to merge, but I will make sure to get this in for the next minor version.

**Comment by aminya** [comment]
@ryansolid I forgot to say. I would like to make this an independent npm package because I use this code in other places. Once you made it working let me know to make an npm package.

**Comment by ryansolid** [comment]
Yeah seems reasonable. I'm already moving it to a separate private package here, I think extracting it out would be beneficial for many.

**Comment by ryansolid** [comment]
Yeah downloading doesn't seem to work locally or in the CI. I get:
```
Error: Cross origin http://localhost forbidden
solid-js:           at dispatchError (/home/runner/work/solid/solid/node_modules/jsdom/lib/jsdom/living/xhr/xhr-utils.js:62:19)
solid-js:           at Object.validCORSHeaders (/home/runner/work/solid/solid/node_modules/jsdom/lib/jsdom/living/xhr/xhr-utils.js:74:5)
```
I've moved the code and set it up I think how I want it for now but detached it from the CI process. This needs more work. If you want to make it standalone go for it. But it looks like whatever issue is affecting the downloads is universal.

**Pull request merged**